ArcGISRuntimeSDKQt_CppSamples/Analysis/FindFeaturesUtilityNetwork/FindFeaturesUtilityNetwork.cpp

```diff
@@ -0,0 +1,304 @@
+        emit terminalDialogVisisbleChanged();
+        return;
+      }
+      else if (m_terminals.size() == 1)
+      {
+        element = m_utilityNetwork->createElementWithArcGISFeature(m_feature, m_terminals[0]);
+      }
+      else
+      {
+        return;
+      }
+
+    }
+    else if (networkSource->sourceType() == UtilityNetworkSourceType::Edge)
+    {
+      element = m_utilityNetwork->createElementWithArcGISFeature(m_feature, nullptr ,this);
+
+      // Compute how far tapped location is along the edge feature.
+      if (m_feature->geometry().geometryType() == GeometryType::Polyline)
+      {
+        const Polyline line = GeometryEngine::removeZ(m_feature->geometry());
+        // Set how far the element is along the edge.
+        element->setFractionAlongEdge(GeometryEngine::fractionAlong(line, m_clickPoint, -1));
+      }
+    }
+    else
+    {
+      return;
+    }
+
+    updateTraceParams(element);
+  });
+}
+
+FindFeaturesUtilityNetwork::~FindFeaturesUtilityNetwork() = default;
+
+void FindFeaturesUtilityNetwork::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<FindFeaturesUtilityNetwork>("Esri.Samples", 1, 0, "FindFeaturesUtilityNetworkSample");
+}
+
+MapQuickView* FindFeaturesUtilityNetwork::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void FindFeaturesUtilityNetwork::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  emit mapViewChanged();
+}
+
+void FindFeaturesUtilityNetwork::multiTerminalIndex(const int &index)
+{
+  if (m_terminals.isEmpty())
+    return;
+  if (!m_feature)
+    return;
+
+  UtilityElement* element;
+  element = m_utilityNetwork->createElementWithArcGISFeature(m_feature, m_terminals[index]);
+  updateTraceParams(element);
+}
+
+void FindFeaturesUtilityNetwork::updateTraceParams(UtilityElement* element)
+{
+  if (startingLocationsEnabled)
+  {
+    m_startingLocations.append(element);
+    m_traceParams->setStartingLocations(m_startingLocations);
+    Graphic* traceLocation = new Graphic(m_clickPoint, m_startingSymbol, this);
+    m_graphicsOverlay->graphics()->append(traceLocation);
+  }
+  else
+  {
+    m_barriers.append(element);
+    m_traceParams->setBarriers(m_barriers);
+    Graphic* traceLocation = new Graphic(m_clickPoint, m_barrierSymbol, this);
+    m_graphicsOverlay->graphics()->append(traceLocation);
+  }
+}
+
+void FindFeaturesUtilityNetwork::trace()
+{
+  busy = true;
+  emit busyChanged();
+  // Perform a connected trace on the utility network
+  m_utilityNetwork->trace(m_traceParams);
+}
+
+void FindFeaturesUtilityNetwork::reset()
+{
+  m_startingLocations.clear();
+  m_traceParams->setStartingLocations(m_startingLocations);
+  m_barriers.clear();
+  m_traceParams->setBarriers(m_barriers);
+  m_graphicsOverlay->graphics()->clear();
```

Comment:
this won't delete graphics, it'll only remove them from the collection. But all of the graphics were created with `this` as parent, so they won't get deleted until the end of the application life cycle. Probably best to go through and delete these, or use some other RAII tactic